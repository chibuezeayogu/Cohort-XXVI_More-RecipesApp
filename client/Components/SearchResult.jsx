import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RecipeCard from './RecipeCard.jsx';
import SmallPreloader from './SmallPreloader.jsx';


class SearchResult extends Component {
  render() {
    const { recipes } = this.props.searchReducer;
    if (!recipes) {
      return (
        <div
          className="row left align-recipe"
          style={{ width: '100%' }}
        >
          <SmallPreloader />
        </div>
      );
    }
    if (recipes && recipes.length === 0) {
      return (
        <div
          id="noSearchFound"
          className="row left"
          style={{ width: '100%' }}
        >
          <h4 className="center-align">No result found</h4>
        </div>
      );
    } else if (recipes && recipes.length > 0) {
      return (
        <div
          className="row left align-recipe"
          style={{ width: '100%' }}
        >
          {recipes ? recipes.map((recipe, i) => (<RecipeCard
            {...this.props}
            key={i}
            recipe={recipe}
          />)) : ''}
        </div>
      );
    }
  }
}

SearchResult.propTypes = {
  searchReducer: PropTypes.shape({
    recipes: PropTypes.shape.isRequired
  }).isRequired
};


export default SearchResult;
