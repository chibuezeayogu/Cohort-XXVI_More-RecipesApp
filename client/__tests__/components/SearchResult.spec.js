import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import SearchResult from '../../Components/SearchResult';
import mockData from '../__mock__/actionMockData';

const props = {
  searchReducer: {
    recipes: mockData.data.recipes,
  }
};

describe('<SearchResult />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SearchResult {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
