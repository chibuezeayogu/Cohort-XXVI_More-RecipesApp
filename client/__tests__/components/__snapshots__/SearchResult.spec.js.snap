// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="row left align-recipe"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <withRouter(RecipeCard)
      recipe={
        Object {
          "addedBy": 1,
          "createdAt": "2018-01-29T11:38:51.147Z",
          "description": "adsfasfa",
          "downvotes": 0,
          "id": 3,
          "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
          "ingredients": "asdfafafa",
          "procedures": "adfadfadf",
          "title": "New title",
          "updatedAt": "2018-02-02T15:30:35.425Z",
          "upvotes": 1,
          "views": 0,
        }
      }
      searchReducer={
        Object {
          "recipes": Array [
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-29T11:38:51.147Z",
              "description": "adsfasfa",
              "downvotes": 0,
              "id": 3,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "asdfafafa",
              "procedures": "adfadfadf",
              "title": "New title",
              "updatedAt": "2018-02-02T15:30:35.425Z",
              "upvotes": 1,
              "views": 0,
            },
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-28T22:46:03.051Z",
              "description": "asdfafd",
              "downvotes": 0,
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "adfadfadf",
              "procedures": "adfafadf",
              "title": "CHANGED",
              "updatedAt": "2018-02-06T19:28:36.149Z",
              "upvotes": 1,
              "views": 0,
            },
          ],
        }
      }
    />
    <withRouter(RecipeCard)
      recipe={
        Object {
          "addedBy": 1,
          "createdAt": "2018-01-28T22:46:03.051Z",
          "description": "asdfafd",
          "downvotes": 0,
          "id": 1,
          "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
          "ingredients": "adfadfadf",
          "procedures": "adfafadf",
          "title": "CHANGED",
          "updatedAt": "2018-02-06T19:28:36.149Z",
          "upvotes": 1,
          "views": 0,
        }
      }
      searchReducer={
        Object {
          "recipes": Array [
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-29T11:38:51.147Z",
              "description": "adsfasfa",
              "downvotes": 0,
              "id": 3,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "asdfafafa",
              "procedures": "adfadfadf",
              "title": "New title",
              "updatedAt": "2018-02-02T15:30:35.425Z",
              "upvotes": 1,
              "views": 0,
            },
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-28T22:46:03.051Z",
              "description": "asdfafd",
              "downvotes": 0,
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "adfadfadf",
              "procedures": "adfafadf",
              "title": "CHANGED",
              "updatedAt": "2018-02-06T19:28:36.149Z",
              "upvotes": 1,
              "views": 0,
            },
          ],
        }
      }
    />
  </div>,
  "nodes": Array [
    <div
      className="row left align-recipe"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <withRouter(RecipeCard)
        recipe={
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-29T11:38:51.147Z",
            "description": "adsfasfa",
            "downvotes": 0,
            "id": 3,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "asdfafafa",
            "procedures": "adfadfadf",
            "title": "New title",
            "updatedAt": "2018-02-02T15:30:35.425Z",
            "upvotes": 1,
            "views": 0,
          }
        }
        searchReducer={
          Object {
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
      />
      <withRouter(RecipeCard)
        recipe={
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-28T22:46:03.051Z",
            "description": "asdfafd",
            "downvotes": 0,
            "id": 1,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "adfadfadf",
            "procedures": "adfafadf",
            "title": "CHANGED",
            "updatedAt": "2018-02-06T19:28:36.149Z",
            "upvotes": 1,
            "views": 0,
          }
        }
        searchReducer={
          Object {
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchResult
        searchReducer={
          Object {
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchResult {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "searchReducer": Object {
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="row left align-recipe"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <withRouter(RecipeCard)
            recipe={
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              }
            }
            searchReducer={
              Object {
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-29T11:38:51.147Z",
                    "description": "adsfasfa",
                    "downvotes": 0,
                    "id": 3,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "asdfafafa",
                    "procedures": "adfadfadf",
                    "title": "New title",
                    "updatedAt": "2018-02-02T15:30:35.425Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf",
                    "procedures": "adfafadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
          />
          <withRouter(RecipeCard)
            recipe={
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              }
            }
            searchReducer={
              Object {
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-29T11:38:51.147Z",
                    "description": "adsfasfa",
                    "downvotes": 0,
                    "id": 3,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "asdfafafa",
                    "procedures": "adfadfadf",
                    "title": "New title",
                    "updatedAt": "2018-02-02T15:30:35.425Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf",
                    "procedures": "adfafadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="row left align-recipe"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <withRouter(RecipeCard)
            recipe={
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              }
            }
            searchReducer={
              Object {
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-29T11:38:51.147Z",
                    "description": "adsfasfa",
                    "downvotes": 0,
                    "id": 3,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "asdfafafa",
                    "procedures": "adfadfadf",
                    "title": "New title",
                    "updatedAt": "2018-02-02T15:30:35.425Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf",
                    "procedures": "adfafadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
          />
          <withRouter(RecipeCard)
            recipe={
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              }
            }
            searchReducer={
              Object {
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-29T11:38:51.147Z",
                    "description": "adsfasfa",
                    "downvotes": 0,
                    "id": 3,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "asdfafafa",
                    "procedures": "adfadfadf",
                    "title": "New title",
                    "updatedAt": "2018-02-02T15:30:35.425Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf",
                    "procedures": "adfafadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchResult
    searchReducer={
      Object {
        "recipes": Array [
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-29T11:38:51.147Z",
            "description": "adsfasfa",
            "downvotes": 0,
            "id": 3,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "asdfafafa",
            "procedures": "adfadfadf",
            "title": "New title",
            "updatedAt": "2018-02-02T15:30:35.425Z",
            "upvotes": 1,
            "views": 0,
          },
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-28T22:46:03.051Z",
            "description": "asdfafd",
            "downvotes": 0,
            "id": 1,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "adfadfadf",
            "procedures": "adfafadf",
            "title": "CHANGED",
            "updatedAt": "2018-02-06T19:28:36.149Z",
            "upvotes": 1,
            "views": 0,
          },
        ],
      }
    }
  />,
}
`;
